wave_lang/kernel/wave/templates/extend_attention.py

```diff
@@ -130,72 +130,20 @@ def get_extend_attention_kernel(
         )
     ]
 
-    i = tkw.IndexMapping.iterator(0)
-    j = tkw.IndexMapping.iterator(1)
-    k = tkw.IndexMapping.iterator(2)
-    d0 = tkw.IndexMapping.dynamic_val(0)
-
-    mapping = tkw.IndexMapping(
-        num_iterators=3,
-        inputs={H: i, D_KV: j, N_Q: k},
-        outputs={H: i, N_Q: k + EXT_IDX, D_KV: j},
-    )
-
-    q_mapping = tkw.IndexMapping(
-        num_iterators=3,
-        inputs={H: i, N_Q: j + EXT_IDX, D_Q: k},
-        outputs={H: i, N_Q: j, D_Q: k},
-    )
+    # Define iterator symbols and bind them to the corresponding dimensions.
+    h_kv = tkl.sym.h_kv
+    d_kv = tkl.sym.d_kv
+    n_kv = tkl.sym.n_kv
+    h = tkl.sym.h
+    n_q = tkl.sym.n_q
+    d_q = tkl.sym.d_q
+    s = tkl.sym.s
+
+    bindings = {h_kv: H_KV, d_kv: D_KV, n_kv: N_KV}
+    bindings.update({h: H, n_q: N_Q, d_q: D_Q, s: S})
```

Comment:
Can all these variables be defined as in line 142? Is `update` being used just for completeness/cleaner code?

Reply:
yes, just wanted to separate variables for *_KV and *_Q, but can all be done in one line.